app/gui/src/dashboard/pages/dashboard/Dashboard.tsx

```diff
@@ -318,3 +328,116 @@ function DashboardInner(props: DashboardProps) {
+              backend: remoteBackend,
+              parentId: project.parentId,
+              category,
+            }),
+          ),
+        )
+        void Promise.allSettled(promises)
+          .then((projects) =>
+            projects.flatMap((directoryResult, i) => {
+              const projectInfo = relevantProjects[i]
+              const project =
+                projectInfo && directoryResult.status === 'fulfilled' ?
+                  directoryResult.value
+                    .filter(backendModule.assetIsProject)
+                    .find((asset) => asset.id === projectInfo.id)
+                : null
+              return project ? [project] : []
+            }),
+          )
+          .then((projects) => {
+            const expandedDirectoryIds = new Set(driveStore.getState().expandedDirectoryIds)
+            for (const project of projects) {
+              const parents = project.parentsPath.split('/').map(backendModule.DirectoryId)
+              const rootDirectoryId = parents[0]
+              if ((rootDirectoryId && userGroupDirectoryIds.has(rootDirectoryId)) ?? false) {
+                expandedDirectoryIds.add(TEAMS_DIRECTORY_ID)
+              } else {
+                expandedDirectoryIds.add(USERS_DIRECTORY_ID)
+              }
+              for (const parent of parents) {
+                expandedDirectoryIds.add(parent)
+              }
+            }
+            setExpandedDirectoryIds([...expandedDirectoryIds])
+          })
+        break
+      }
+      case 'local':
+      case 'local-directory': {
+        if (!localBackend) {
+          break
+        }
+        const rootPath = 'rootPath' in category ? category.rootPath : localBackend.rootPath()
+        const relevantProjects = launchedProjectsRef.current.filter(
+          (project) => project.type === backendModule.BackendType.local,
+        )
+        const expandedDirectoryIds = new Set(driveStore.getState().expandedDirectoryIds)
+        for (const project of relevantProjects) {
+          const path = localBackendModule.extractTypeAndId(project.parentId).id
+          const strippedPath = path.replace(`${rootPath}/`, '')
+          if (strippedPath !== path) {
+            let parentPath = String(rootPath)
+            const parents = strippedPath.split('/')
+            for (const parent of parents) {
+              parentPath += `/${parent}`
+              expandedDirectoryIds.add(
+                localBackendModule.newDirectoryId(backendModule.Path(parentPath)),
+              )
+            }
+          }
+        }
+        setExpandedDirectoryIds([...expandedDirectoryIds])
+        break
+      }
+      case 'recent':
+      case 'trash': {
+        // Ignored - directories should not be expanded here.
+        break
+      }
+    }
+  }, [
+    category,
+    driveStore,
+    launchedProjectsRef,
+    localBackend,
+    queryClient,
+    remoteBackend,
+    setExpandedDirectoryIds,
+    user.userGroups,
```

Comment:
@MrFlashAccount not too sure about this one. should i be putting things in a ref like elsewhere, to make sure it doesn't unintentionally open all parent directories out of nowhere?

(is this not the right approach at all?)

Reply:
I don't get why it's an useEffect

Reply:
true, what's the alternative though? i think we need to re-expand the directories every time the category changes, right?

Reply:
we can avoid resetting `expandedDirectories` every time the category changes and only run this one. but i think it'd still need to be a `useEffect`, right? i don't think it should be hardcoded to be permanently expanded because otherwise you wouldn't collapse directories (e.g. if the directory has a lot of children so you want to collapse it)

Reply:
actually, i don't know if this is an option right now as currently we depend on `expandedDirectoryIds` to control which `listDirectory` queries are being fetched

Reply:
Could you elaborate on what we expect? Should this expand only on open? Or should expand once user `somehow` launch a project?

Reply:
only on open i think - @PabloBuchu for confirmation.
although it's true that if we open a project from the "recent" category then the user might want the directory and its parents to be expanded.

although note that opening from recent wouldn't work if we're resetting expanded directories state every time we switch category (which we currently need to do as mentioned above)

Reply:
on that note, for *user actions* specifically we can handle changing the set of expanded directories when pressing the "open project" button anyway

Reply:
> although note that opening from recent wouldn't work if we're resetting expanded directories state every time we switch category (which we currently need to do as mentioned above)

This can be bypassed:https://github.com/enso-org/enso/blob/develop/app/gui/src/dashboard/components/dashboard/column/PathColumn.tsx#L65

Reply:
> only on open i think - @PabloBuchu for confirmation.

Deps in `useEffect`says that every time one of them change, we should expand directories. I'm not sure it's the `desired` behavior...

Reply:
How I see this:

1. Expand on open only
2. Add context menu to the project tabs with option `Show in Drive` (subject of change)

@somebody1234 , @PabloBuchu thoughts?

Reply:
@somebody1234 also I think we can turn this https://github.com/enso-org/enso/blob/develop/app/gui/src/dashboard/components/dashboard/column/PathColumn.tsx#L51 into a utility as I think you do similar things in your PR.

Reply:
> oh no, i mean *preserving expanded directories state after opening a project from "recent"

Let's ignore this for now

Reply:
> Deps in `useEffect` says that every time one of them change, we should expand directories

yup. that was my original concern, *but* i don't see a way to avoid it fully. in the react compiler optimization PR i just moved dependencies that aren't *real* dependencies to a syncref but if you have a better idea i'm definitely down to change it

Reply:
> Expand on open only, context menu to open tabs

sounds good, but i'm worried about discoverability. i think it'd be cool for the icons to change into buttons to trigger dropdown menus on hover, but i dunno whether that's a good idea

Reply:
> yup. that was my original concern, but i don't see a way to avoid it fully. in the react compiler optimization PR i just moved dependencies that aren't real dependencies to a syncref but if you have a better idea i'm definitely down to change it

We have `useEventCallback` for that, here's a good explanation on why it's needed: https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md

Reply:
> sounds good, but i'm worried about discoverability. i think it'd be cool for the icons to change into buttons to trigger dropdown menus on hover, but i dunno whether that's a good idea

Yeah, good idea

Reply:
rfe: useEventCallback... thoughts? do you mean to reference non-deps in the eventCallback directly and then use a `useEffect` that passes the reactive deps into the function?

Reply:
A question: `when` the useEffect should be triggered by `design`?

Reply:
One thing on implementation: ideally this should support multitabs

Reply:
>  thoughts? do you mean to reference non-deps in the eventCallback directly and then use a useEffect that passes the reactive deps into the function?

Yeah, like that. Deps that shouldn't trigger effect, should be incapsulated inside the `useEventCallback` function